DynamicJsonDocument deprecated in ArduinoJSON ver. 7

The code will not compile without errors In PlatformIO because the DynamicJsonDocument has been deprecated in ArduinoJSON version 7.  For example:
`In file included from .pio/libdeps/lilygo-t-display/ArduinoJson/src/ArduinoJson.hpp:53,
                 from .pio/libdeps/lilygo-t-display/ArduinoJson/src/ArduinoJson.h:9,
                 from src/ConfigurationManagement/configuration.h:9,
                 from src/System/TaskManager.h:9,
                 from src/TaskMQTT.h:4,
                 from src/TaskMQTT.cpp:4:
.pio/libdeps/lilygo-t-display/ArduinoJson/src/ArduinoJson/compatibility.hpp:125:58: note: declared here
 class ARDUINOJSON_DEPRECATED("use JsonDocument instead") DynamicJsonDocument`
or here:
`cc1plus.exe: out of memory allocating 8392703 bytes
src/TaskMQTT.cpp: In member function 'virtual bool MQTTTask::loop(System&)':     
src/TaskMQTT.cpp:32:29: warning: 'DynamicJsonDocument' is deprecated: use JsonDocument instead [-Wdeprecated-declarations]
     DynamicJsonDocument data(1024);`

Workarounds? Or upgrade code? 

 - OS: Win10.0.19045
 - PlatformIO Version: 6.1.13 / 3.4.4
 - VSC: 1.87.2
 - Board: LilyGo T3 V1.6.1

--- a/src/ConfigurationManagement/configuration.cpp
+++ b/src/ConfigurationManagement/configuration.cpp
@@ -23,7 +23,7 @@ void ConfigurationManagement::readConfiguration(logging::Logger &logger, Configu
     logger.log(logging::LoggerLevel::LOGGER_LEVEL_ERROR, MODULE_NAME, "Failed to open file for reading, using default configuration.");
     return;
   }
-  JsonDocument  data;
+  JsonDocument         data;
   DeserializationError error = deserializeJson(data, file);
   if (error) {
     logger.log(logging::LoggerLevel::LOGGER_LEVEL_WARN, MODULE_NAME, "Failed to read file, using default configuration.");

--- a/src/ConfigurationManagement/configuration.cpp
+++ b/src/ConfigurationManagement/configuration.cpp
@@ -23,7 +23,7 @@ void ConfigurationManagement::readConfiguration(logging::Logger &logger, Configu
     logger.log(logging::LoggerLevel::LOGGER_LEVEL_ERROR, MODULE_NAME, "Failed to open file for reading, using default configuration.");
     return;
   }
-  DynamicJsonDocument  data(2048);
+  JsonDocument  data;
   DeserializationError error = deserializeJson(data, file);
   if (error) {
     logger.log(logging::LoggerLevel::LOGGER_LEVEL_WARN, MODULE_NAME, "Failed to read file, using default configuration.");
@@ -44,7 +44,7 @@ void ConfigurationManagement::writeConfiguration(logging::Logger &logger, Config
     logger.log(logging::LoggerLevel::LOGGER_LEVEL_ERROR, MODULE_NAME, "Failed to open file for writing...");
     return;
   }
-  DynamicJsonDocument data(2048);
+  JsonDocument data;
 
   writeProjectConfiguration(conf, data);
 

--- a/src/ConfigurationManagement/configuration.h
+++ b/src/ConfigurationManagement/configuration.h
@@ -22,8 +22,8 @@ public:
   void writeConfiguration(logging::Logger &logger, Configuration &conf);
 
 private:
-  virtual void readProjectConfiguration(DynamicJsonDocument &data, Configuration &conf)  = 0;
-  virtual void writeProjectConfiguration(Configuration &conf, DynamicJsonDocument &data) = 0;
+  virtual void readProjectConfiguration(JsonDocument &data, Configuration &conf)  = 0;
+  virtual void writeProjectConfiguration(Configuration &conf, JsonDocument &data) = 0;
 
   const String mFilePath;
 };

--- a/src/TaskMQTT.cpp
+++ b/src/TaskMQTT.cpp
@@ -29,7 +29,7 @@ bool MQTTTask::loop(System &system) {
   if (!_toMQTT.empty()) {
     std::shared_ptr<APRSMessage> msg = _toMQTT.getElement();
 
-    DynamicJsonDocument data(1024);
+    JsonDocument data;
     data["source"]      = msg->getSource();
     data["destination"] = msg->getDestination();
     data["path"]        = msg->getPath();

--- a/src/project_configuration.cpp
+++ b/src/project_configuration.cpp
@@ -4,7 +4,7 @@
 
 #include "project_configuration.h"
 
-void ProjectConfigurationManagement::readProjectConfiguration(DynamicJsonDocument &data, Configuration &conf) {
+void ProjectConfigurationManagement::readProjectConfiguration(JsonDocument &data, Configuration &conf) {
   if (data.containsKey("callsign"))
     conf.callsign = data["callsign"].as<String>();
 
@@ -119,7 +119,7 @@ void ProjectConfigurationManagement::readProjectConfiguration(DynamicJsonDocumen
     conf.board = data["board"].as<String>();
 }
 
-void ProjectConfigurationManagement::writeProjectConfiguration(Configuration &conf, DynamicJsonDocument &data) {
+void ProjectConfigurationManagement::writeProjectConfiguration(Configuration &conf, JsonDocument &data) {
   data["callsign"] = conf.callsign;
 
   if (!conf.network.DHCP) {

--- a/src/project_configuration.cpp
+++ b/src/project_configuration.cpp
@@ -134,9 +134,9 @@ void ProjectConfigurationManagement::writeProjectConfiguration(Configuration &co
   }
 
   data["wifi"]["active"] = conf.wifi.active;
-  JsonArray aps          = data["wifi"].createNestedArray("AP");
+  JsonArray aps          = data["wifi"]["AP"].to<JsonArray>();
   for (Configuration::Wifi::AP ap : conf.wifi.APs) {
-    JsonObject v  = aps.createNestedObject();
+    JsonObject v  = aps.add<JsonObject>();
     v["SSID"]     = ap.SSID;
     v["password"] = ap.password;
   }
@@ -165,9 +165,9 @@ void ProjectConfigurationManagement::writeProjectConfiguration(Configuration &co
   data["display"]["overwrite_pin"]        = conf.display.overwritePin;
   data["display"]["turn180"]              = conf.display.turn180;
   data["ftp"]["active"]                   = conf.ftp.active;
-  JsonArray users                         = data["ftp"].createNestedArray("user");
+  JsonArray users                         = data["ftp"]["user"].to<JsonArray>();
   for (Configuration::Ftp::User u : conf.ftp.users) {
-    JsonObject v  = users.createNestedObject();
+    JsonObject v  = users.add<JsonObject>();
     v["name"]     = u.name;
     v["password"] = u.password;
   }

--- a/src/project_configuration.h
+++ b/src/project_configuration.h
@@ -176,8 +176,8 @@ public:
   }
 
 private:
-  virtual void readProjectConfiguration(DynamicJsonDocument &data, Configuration &conf) override;
-  virtual void writeProjectConfiguration(Configuration &conf, DynamicJsonDocument &data) override;
+  virtual void readProjectConfiguration(JsonDocument &data, Configuration &conf) override;
+  virtual void writeProjectConfiguration(Configuration &conf, JsonDocument &data) override;
 };
 
 #endif

